image/templates/helm/stackrox-secured-cluster/templates/sensor.yaml.htpl

```diff
@@ -313,6 +313,6 @@ metadata:
     {{- include "srox.annotations" (list . "secret" "helm-effective-cluster-name" $annotations) | nindent 4 }}
 stringData:
   cluster-name: |
-    {{- ._rox.clusterName | nindent 4 }}
+    {{- printf "%s" (._rox.clusterName | toString) | nindent 4 }}
```

Comment:
I don't think this is the right place to stringify the cluster name. I suspect it's already too late - from your testing it seems a large number provided by the user is already converted (in a lossy manner) to a float at this point 🤔

Reply:
Conversion to string works as the error from the description does not appear anymore for low values (123). Only the conversion to scientific notation suggests that earlier before this step the number is manipulated as a number, not string.

Reply:
This looks like not a trivial fix. For now, I decided to reject cluster names that are parsable to a number - see [cd208e7](https://github.com/stackrox/stackrox/pull/12295/commits/cd208e768b766c9be8458ddf4cd4457643bb5f70)